assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -3404,7 +3404,9 @@ protected String build_instructor_grade_submission_context(VelocityPortlet portl
         }
         context.put("alertGradeDraft", Boolean.valueOf(addGradeDraftAlert));
 
-        rangeAndGroups.buildInstructorGradeSubmissionContextGroupCheck(assignment, submission.get().getGroupId(), state);
+        if (submission.isPresent()) {
+            rangeAndGroups.buildInstructorGradeSubmissionContextGroupCheck(assignment, submission.get().getGroupId(), state);
+        }
```

Comment:
This can be shortened to:
`submission.ifPresent(s -> rangeAndGroups.buildInstructorGradeSubmissionContextGroupCheck(assignment, s.getGroupId(), state));`

Reply:
Since there already is a section that checks the existence of the submission can this be added there?

Reply:
Yeah, I assume I thought this code was already in that block when I wrote it. That's the problem with methods that are hundreds of lines long. I'll move it up to the bottom of that block.